flow/activities/flowable_core.go

```diff
@@ -144,7 +144,8 @@ func syncCore[TPull connectors.CDCPullConnectorCore, TSync connectors.CDCSyncCon
 	lastOffset, err := func() (model.CdcCheckpoint, error) {
 		// special case pg-pg replication, where offsets are stored on destination instead of catalog
 		if _, isPg := any(srcConn).(*connpostgres.PostgresConnector); isPg {
-			dstPgConn, err := connectors.GetByNameAs[*connpostgres.PostgresConnector](ctx, config.Env, a.CatalogPool, config.DestinationName)
+			dstPgConn, err := connectors.GetByNameAndForPeerAs[*connpostgres.PostgresConnector](
```

Comment:
Type parameter must always be same with last parameter. Better to create specific (non generic) function for postgres. Or switch over type parameter